common/src/main/java/com/microsoft/hydralab/performance/parsers/AndroidMemoryInfoResultParser.java

```diff
@@ -0,0 +1,141 @@
+            long[] memInfos = parseRawResultFile(logFile);
+            inspectionResult.parsedData = buildMemoryInfo(inspectionResult.inspection.appId, inspectionResult.inspection.description, inspectionResult.timestamp, memInfos);
+        }
+
+        return performanceTestResult;
+    }
+
+
+    private AndroidMemoryInfo buildMemoryInfo(String packageName, String description, long timestamp, long[] memInfos) {
+        if (memInfos == null || memInfos.length != 19) return null;
+        AndroidMemoryInfo androidMemoryInfo = new AndroidMemoryInfo();
+        androidMemoryInfo.setAppPackageName(packageName);
+        androidMemoryInfo.setDescription(description);
+        androidMemoryInfo.setTimeStamp(timestamp);
+        androidMemoryInfo.setJavaHeapPss(memInfos[0]);
+        androidMemoryInfo.setJavaHeapRss(memInfos[1]);
+        androidMemoryInfo.setNativeHeapPss(memInfos[2]);
+        androidMemoryInfo.setNativeHeapRss(memInfos[3]);
+        androidMemoryInfo.setCodePss(memInfos[4]);
+        androidMemoryInfo.setCodeRss(memInfos[5]);
+        androidMemoryInfo.setStackPss(memInfos[6]);
+        androidMemoryInfo.setStackRss(memInfos[7]);
+        androidMemoryInfo.setGraphicsPss(memInfos[8]);
+        androidMemoryInfo.setGraphicsRss(memInfos[9]);
+        androidMemoryInfo.setPrivateOtherPss(memInfos[10]);
+        androidMemoryInfo.setPrivateOtherRss(memInfos[11]);
+        androidMemoryInfo.setSystemPss(memInfos[12]);
+        androidMemoryInfo.setSystemRss(memInfos[13]);
+        androidMemoryInfo.setUnknownPss(memInfos[14]);
+        androidMemoryInfo.setUnknownRss(memInfos[15]);
+        androidMemoryInfo.setTotalPss(memInfos[16]);
+        androidMemoryInfo.setTotalRss(memInfos[17]);
+        androidMemoryInfo.setTotalSwapPss(memInfos[18]);
+        return androidMemoryInfo;
+    }
+
+    private long[] parseRawResultFile(File rawFile) {
+        String line;
+        long[] memoryValueArr = new long[19];
+        try (BufferedReader bufferedReader = new BufferedReader(new FileReader(rawFile))) {
+            while ((line = bufferedReader.readLine()) != null) {
+                if (line.trim().startsWith("App Summary")) {
+                    // PSS title line, used to anchor target index offset
+                    line = bufferedReader.readLine();
+                    int pssEndOffset = line.indexOf(")");
+                    int rssEndOffset = line.lastIndexOf(")");
+
+                    // move to data line
+                    bufferedReader.readLine();
+                    line = bufferedReader.readLine();
+                    while (line != null) {
+                        String lineStr = line.trim();
+
+                        if (lineStr.startsWith("TOTAL ")) {
+                            if (line.contains("TOTAL PSS:")) {
+                                String pssValue = line.split("TOTAL PSS:")[1].split(" +")[1];
+                                memoryValueArr[16] = Integer.parseInt(pssValue);
```

Comment:
Do we need to catch `NumberFormatException`?

Reply:
Nice catch, fixed.